src/makeRobot.js

```diff
@@ -38,7 +38,76 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
-}
+  const robot = {
+    goRight: goRight,
+    goLeft: goLeft,
+    goForward: goForward,
+    goBack: goBack,
+    evacuate: evacuate,
+
+    name,
+    wheels,
+    version,
+
+    coords: {
+      'x': 0,
+      'y': 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, chip version: ${this.version},`
+      + ` wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${robot.coords['x']}, y=${robot.coords['y']}`;
+    },
+  };
+
+  function goRight(step = 1) {
+    if (step < 0) {
+      return robot;
+    }
+    robot.coords['x'] += step;
+
+    return robot;
```

Comment:
You can rewrite like this everywhere
```suggestion
    if (step > 0) {
      this.coords.x += step;
    }

    return this;
```